Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp

```diff
@@ -200,11 +200,12 @@ void AsyncScrollingCoordinator::frameViewWillBeDetached(FrameView& frameView)
     if (!coordinatesScrollingForFrameView(frameView))
         return;
 
-    auto* scrollingNode = downcast<ScrollingStateScrollingNode>(m_scrollingStateTree->stateNodeForID(frameView.scrollingNodeID()));
-    if (!scrollingNode)
+    auto scrollingNode = m_scrollingStateTree->stateNodeForID(frameView.scrollingNodeID());
+    auto* node = scrollingNode ? dynamicDowncast<ScrollingStateScrollingNode>(*scrollingNode) : nullptr;
```

Comment:
No need to null check and then use `dynamicDowncast<>`; just use the pointer version. So this could just be:
```
auto* stateNode = dynamicDowncast<ScrollingStateScrollingNode>(m_scrollingStateTree->stateNodeForID(frameView.scrollingNodeID()));
```

Maybe we could even make a template helper for this pattern, which will be very common.

Reply:
I tried that already. It doesn't compile.

Reply:
Actually, I don't want to do that anyway because I want to have a Ref/RefPtr in stack instead of just a raw pointer.